module/localisation/BallLocalisation/src/BallModel.hpp

```diff
@@ -38,37 +44,54 @@ namespace localisation {
 
         static constexpr size_t size = 2;
 
+        using StateVec = Eigen::Matrix<Scalar, size, 1>;
+        using StateMat = Eigen::Matrix<Scalar, size, size>;
+
         struct MeasurementType {
             struct BALL {};
         };
 
-        arma::vec2 processNoiseDiagonal;
-
+        // Local variables for this model. Set their values from config file
+        // Number of reset particles
+        int NUM_ROGUES;
+        // Range of reset particles
+        Eigen::Matrix<Scalar, 2, 1> resetRange;
+        // Diagonal noise matrix
+        Eigen::Matrix<Scalar, 2, 1> processNoiseDiagonal;
 
-        BallModel() : processNoiseDiagonal(arma::fill::eye) {}  // empty constructor
+        BallModel() : NUM_ROGUES(), resetRange(), processNoiseDiagonal() {}
```

Comment:
I removed those initialisations because on configuration those reactors get those values from config files. It seems to me like all that initializing those here would do is give you the wrong impression that those initialisation values actually do something which affects the reactor.
Maybe I am overthinking it...

Reply:
I think explicit default initialization should still exist. Without the explicit initializes you are left with implicit default initialized variables, which for eigen types is likely to be uninitilized memory